Support demangling symbols with dot-delimited words at the end

shepmaster opened the pull request:
This accounts for values like LLVM IR branch labels.

Closes #15

alexcrichton commented:
Nice! Could some validation also be added that everything after the E. is "symbol like" to avoid eagerly desymbolizing things that shouldn't be?

shepmaster commented:
> Could some validation also be added that everything after the E. is "symbol like" to avoid eagerly desymbolizing things that shouldn't be?

I've updated to make sure the suffix is ASCII alphanumeric + ASCII punctuation and added a test.

alexcrichton commented:
👍 

Looks like CI may be failing though?

shepmaster commented:
> Looks like CI may be failing though?

Huh...

```
error[E0599]: no method named `is_alphanumeric` found for type `char` in the current scope
   --> src/lib.rs:224:11
    |
224 |         c.is_alphanumeric() || is_ascii_punctuation(c)
    |           ^^^^^^^^^^^^^^^
```

but [`char::is_alphanumeric`](https://doc.rust-lang.org/std/primitive.char.html#method.is_alphanumeric) is since 1.0:

```rust
    #[stable(feature = "rust1", since = "1.0.0")]
    #[inline]
    pub fn is_alphanumeric(self) -> bool {
        self.is_alphabetic() || self.is_numeric()
    }
```

Am I asleep? Am I missing something really obvious?

shepmaster commented:
Oh, `no_std`. Ok, I'll copy that implementation too.

shepmaster commented:
Thank you!
